ui/src/components/UserForm.js

```diff
@@ -95,6 +101,43 @@ export class UserForm extends LitElement {
         saveButton.disabled = !formValidity;
     }
 
+    /**
+     * Enables the save button when accessed via the login flow if
+     * the cookies and URL contain every required piece of user information.
+     */
+    checkInitialEnable() {
+        let formValidity = true;
+        const filledOut = 
+            this.userId && 
+            this.gender && 
+            this.userAge && 
+            this.deviceType
+
+        if (!filledOut) {
+            formValidity = false; 
+        }
+
+        if (this.userAge < MIN_USER_AGE 
+            || this.userAge > MAX_USER_AGE) {
+                formValidity = false; 
+        }
+
+        const saveButton = this.shadowRoot.getElementById('save-button');
+        saveButton.disabled = !formValidity;
```

Comment:
use the templates to update component **properties:**

this.disableSaveButton = !formValidity;

and in the save button template:
?disabled=${!this.disableSaveButton}

----
another option is to set a form validity property.
instead of checkInitialEnable, rename to checkFormValidity and set this.formValidity.

and in. the save button template:
?disabled=${!this.loginCompleted && !this.formValidity}

Reply:
adjusted to first option